### DISCUSSION TITLE
2TX Taproot p2p trading pipeline

### DISCUSSION THREAD

**Discussion by f321x** [bisq-network/bisq2#2783] [General]
We've been working on a taproot p2p trading pipeline over at the RoboSats project which could possibly be used in Bisq too. It needs 2 onchain transactions to finish the trade. The current state is a working demonstrator in Rust but no production ready library/application yet. You can have a look at the pipeline in the README.md of the Repo https://github.com/RoboSats/taptrade-core. Would this fit in your architecture/be of interest to the Bisq project? If so we could consider design choices in further development to allow for simpler integration in Bisq too.

**Comment by HenrikJannsen** [comment]
Thanks for reaching out! 
A proposal with a MuSig based trade-protocol has been recently added here: https://github.com/bisq-network/proposals/issues/456

I guess it shares some basic ideas.

I just read quickly your readme, so apologize if I misunderstood anything but to me the central role of a coordinator seems problematic from both security, scalability and regulatory point of view. If for dispute resolution no other solution than a 3rd party can be found it might be acceptable but I would not extend its role more as absolutely needed.
Also the financial commitment for posting offers might be a problem to get a market bootstrapped. Bisq has over time developed solutions to reduce the liquidity requirements for makers as that was a pain point for power traders.
I would recommend to look into alternative, non-financial means how to protect against spam or other abuse (e.g. proof or work, reputation,...).

**Reply by f321x** [reply]
Thanks for the feedback, I see the projects have a quite different architecture and risk assumptions. Our trade pipeline in this form seems more useful for hosted exchanges (like Robosats). Also good point re bond requirements👀

**Comment by ChrisSon15** [comment]
Thanks for letting us know of your efforts! Actually there are quite a number of common idea, which we were thinking of. Also I like that you guys are using MuSig2 to add more privacy to the happy path of the protocol.
We are also considering Rust so we actually might have some sharing of code as well. Since we are all up for the same goals like privacy, censorship resistance, etc. I am all up to sharing ideas, code and whatever for reaching our common goal.
I will have a closer technical look at it these days. thanks again for contacting.

**Reply by f321x** [reply]
Cool, didn't know you also consider using Rust. In case you decide for Rust the demonstrator code could be helpful figuring some less documented stuff out (like using musig with bdk), looking forward to bisq's protocol implementation. Feel free to [reach out](https://simplex.chat/contact#/?v=2-5&smp=smp%3A%2F%2FenEkec4hlR3UtKx2NMpOUK_K4ZuDxjWBO1d9Y4YXVaA%3D%40smp14.simplex.im%2Fl9ZKkZObVUivB5n0xviB9Rjdl2t_him7%23%2F%3Fv%3D1-2%26dh%3DMCowBQYDK2VuAyEA-hV3pNDT2b1DZWoIyJ4NLFMTFyMnTTL-RzUFMzPQ9Qk%253D%26srv%3Daspkyu2sopsnizbyfabtsicikr2s4r3ti35jogbcekhm3fsoeyjvgrid.onion&data=%7B%22type%22%3A%22group%22%2C%22groupLinkId%22%3A%22ANU9ku4FkNjsggzUTmpYsQ%3D%3D%22%7D) if you have any questions regarding the protocol/code.
